quickfixj-core/src/main/java/quickfix/Message.java

```diff
@@ -733,7 +733,9 @@ private void parseGroup(String msgType, StringField field, DataDictionary dd, Da
                 break;
             }
             int tag = field.getTag();
-            if (tag == firstField) {
+            boolean shouldCreateNewGroup = tag == firstField || (groupDataDictionary.isFirstFieldInGroupIsDelimiter() && firstField < 0);
```

Comment:
```suggestion
            boolean shouldCreateNewGroup = tag == firstField || (groupDataDictionary.isFirstFieldInGroupIsDelimiter() && firstField == -1);
```
Maybe it would make sense to check for `firstField == -1` here. Obviously `<0` is similar here but checking for `-1` would be the same as in line 726 and hence easier to understand:
```
int firstField = groupDataDictionary.isFirstFieldInGroupIsDelimiter() ? -1 : rg.getDelimiterField();
```

Reply:
Thanks, applied.